test/scales/tickGeneratorsTests.ts

```diff
@@ -22,35 +23,62 @@ describe("Scales", () => {
       });
 
       it("generates ticks for a domain crossing 0", () => {
-        let start = -1.5, end = 1, interval = 0.5;
+        let start = -1.5;
+        let end = 1;
         scale.domain([start, end]);
-        let ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(interval)(scale);
+        let ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(0.5)(scale);
         assert.deepEqual(ticks, [-1.5, -1, -0.5, 0, 0.5, 1], "generated all number divisible by 0.5 in domain");
       });
 
       it("generates ticks with reversed domain", () => {
-        let start = -2.2, end = -7.6, interval = 2.5;
+        let start = -2.2;
+        let end = -7.6;
         scale.domain([start, end]);
-        let ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(interval)(scale);
+        let ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(2.5)(scale);
         assert.deepEqual(ticks, [-7.6, -7.5, -5, -2.5, -2.2], "generated all ticks between lower and higher value");
       });
 
-      it("only returns the ends of the domain if interval is bigger than the domain", () => {
-        let start = 0.5, end = 10.01, interval = 11;
+      it("returns the ends of the domain if interval is NaN or bigger than the domain", () => {
+        let start = 0.5;
+        let end = 10.01;
         scale.domain([start, end]);
-        let ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(interval)(scale);
-        assert.deepEqual(ticks, [0.5, 10.01], "no middle ticks were added");
+        let ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(11)(scale);
+        assert.deepEqual(ticks, [0.5, 10.01], "no middle ticks were added when interval is 11");
+
+        ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(Infinity)(scale);
+        assert.deepEqual(ticks, [0.5, 10.01], "no middle ticks were added when interval is Infinity");
+
+        ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(NaN)(scale);
+        assert.deepEqual(ticks, [0.5, 10.01], "no middle ticks were added when interval is NaN");
+      });
+
+      it("works for Scales.ModifiedLog", () => {
+        let logScale = new Plottable.Scales.ModifiedLog(2);
+        logScale.domain([0, 100]);
+        let logTicks = Plottable.Scales.TickGenerators.intervalTickGenerator(40)(logScale);
+        assert.deepEqual(logTicks, [0, 40, 80, 100], "generates ticks for Scales.ModifiedLog");
       });
 
-      it("rejects negative values", () => {
+      it("rejects non-positive values", () => {
         (<any>assert).throws(() => Plottable.Scales.TickGenerators.intervalTickGenerator(0), Error, "interval must be positive number",
-          "interval must be positive number");
+          "interval cannot be 0");
         (<any>assert).throws(() => Plottable.Scales.TickGenerators.intervalTickGenerator(-2), Error, "interval must be positive number",
```

Comment:
HACKHACK's are mainly used if there is a decently awkward workaround for something that should be inherent.  In this case, we should be very rarely any-casting and especially in this case, we are any-casting since our d.ts file does not have the proper typing definition.

I think I do agree that using this signature is a best practice, but usage of it is a HACKHACK until we fix the typings file.  In this case, we should definitely have a HACKHACK comment everywhere we see this.